Improve contribution message about adding e2e tests (#4805)
crossplane/crossplane: .github/PULL_REQUEST_TEMPLATE.md

I have:

- [ ] Read and followed Crossplane's [contribution process].

> - [ ] Added or updated unit **and** checked with reviewers/maintainers E2E tests are needed for my change.

Reviewer: ```suggestion
- [ ] Added or updated unit tests **and** checked with reviewers/maintainers whether E2E tests are needed for my change.
```

> - [ ] Added or updated unit **and** checked with reviewers/maintainers E2E tests are needed for my change.

Reviewer 2: ```suggestion
- [ ] Added or updated unit tests **and** checked with reviewers/maintainers to determine if E2E tests are needed for my change.
```

> - [ ] Added or updated unit tests **and** checked with reviewers/maintainers whether E2E tests are needed for my change.

Reviewer 3: ```suggestion
- [ ] Added or updated unit tests for my change.
- [ ] Added or updated E2E tests for my change.
```

I'd prefer to break these out into two lines. This is partly because of aesthetics - I think the list looks cleaner when it doesn't wrap. More importantly though, this allows a reviewer to see at a glance whether the PR has unit tests _and_ E2E tests, or only one.

Perhaps we could add a sentence or two to the comment block above to _briefly_ explain on our expectations around unit and E2E tests. That keeps the list uncluttered and easy to read, but still gives folks guidance at a glance.

I have:

- [ ] Read and followed Crossplane's [contribution process].
- [ ] Added or updated unit tests.

> - [ ] Checked with reviewers/maintainers whether E2E tests are needed for my change

Reviewer 4: Checking with reviewers/maintainers should not be needed for obvious cases. Only do that when you're unsure.

```suggestion
- [ ] Added or updated e2e tests (check with reviewers/maintainers if you're unsure whether E2E tests are necessary for the change).
```
